R/workbook.readNamedRegion.R

```diff
@@ -41,20 +41,27 @@ setMethod("readNamedRegion",
 	function(object, name, header = TRUE, rownames = NULL, colTypes = character(0),
 			forceConversion = FALSE, dateTimeFormat = getOption("XLConnect.dateTimeFormat"),
 			check.names = TRUE, useCachedValues = FALSE, keep = NULL, drop = NULL, simplify = FALSE,
-      readStrategy = "default") {
+      readStrategy = "default", worksheetName = .jnull(class = "java/lang/String")) {
 
 		# returns a list of RDataFrameWrapper Java object references
-		sheet = as.vector(extractSheetName(getReferenceFormula(object, name)))
-		namedim = matrix(as.vector(t(getReferenceCoordinatesForName(object, name))), nrow=4, byrow=FALSE)
+	  if(identical(worksheetName,.jnull(class = "java/lang/String"))) {
+		  sheet = as.vector(extractSheetName(getReferenceFormula(object, name)))
+	  }
+	  else {
+	    sheet = as.vector(worksheetName)
+	  }
```

Comment:
The `sheet` variable defined here is use to target subsets of columns in the data via `getColSubset()`. As such it should always be the sheet where the data are, regardless of the scope where the name is defined., 

I guess we need to extend `getReferenceFormula()` to gain the worksheet scope argument and just use it as

```r
sheet = as.vector(extractSheetName(getReferenceFormula(object, name, worksheetName)))
```